mods/parsoid.js

```diff
@@ -444,6 +444,16 @@ PSP.transformRevision = function (restbase, req, from, to) {
 
     return this._getOriginalContent(restbase, req, rp.revision, tid)
     .then(function (original) {
+        // Check if parsoid metadata is present as it's required by parsoid.
+        if (!original['data-parsoid'].body || !original['data-parsoid'].body.ids) {
```

Comment:
It turns out that the corruption is caused by the fact that some objects are being stored and retrieved as strings instead of JSONs. So, this line needs to be:

```js
if (!original['data-parsoid'].body || original['data-parsoid'].body.constructor !== Object || !original['data-parsoid'].body.ids)
```